### PR TITLE
Add section about MATLAB testing to the developers documentation

### PR DESCRIPTION
**Pull request by sbesson**
Document requirements (xUnit framework) and commands to run the MATLAB tests

/cc @bramalingam @manics

### CODE REVIEW

File: docs/sphinx/developers/commit-testing.txt [hunk 1]
```diff
@@ -98,3 +101,29 @@ If Ant reports that the build was successful, then there is nothing that
...
+link above from the MATLAB Central File Exchange. After adding the xUnit
+framework and the MATLAB toolbox to your MATLAB path, you can run the tests
+using:
+
+.. code-block:: matlab
+
+  runtests components/formats-gpl/test/matlab
+
+Alternatively, you can download `matlab-xunit <https://github.com/psexton/matlab-xunit>`_ which is based on the MATLAB xUnit
+Test Framework and adds JUnit compatible XML output. After adding it to your
```
Comment (mtbc) [inline docs/sphinx/developers/commit-testing.txt]: JUnit-compatible

File: docs/sphinx/developers/commit-testing.txt [hunk 2]
```diff
@@ -98,3 +101,29 @@ If Ant reports that the build was successful, then there is nothing that
...
+MATLAB path,  you can run the tests using:
```
Comment (mtbc) [inline docs/sphinx/developers/commit-testing.txt]: double space

File: docs/sphinx/developers/commit-testing.txt [hunk 3]
```diff
@@ -98,3 +101,47 @@ If Ant reports that the build was successful, then there is nothing that
...
+:sourcedir:`components/formats-gpl/test/matlab`.
+
+To run these tests, you will need to download the xUnit Framework using the
+link above from the MATLAB Central File Exchange. After adding the xUnit
+framework and the MATLAB toolbox to your MATLAB path, you can run the tests
+using:
+
+.. code-block:: matlab
+
+  runtests components/formats-gpl/test/matlab
```
Comment (manics) [inline docs/sphinx/developers/commit-testing.txt]: Note this conflicts (and fails) with the builtin `runtests` in Matlab 2015a.

Comment (sbesson) [inline docs/sphinx/developers/commit-testing.txt]: * sighs * makes sense in some way. What about the matlab-xunit approach below?

**Comment by manics** [comment]
From the top level directory:
```
addpath ~/work/bioformats/components/formats-gpl/matlab/
addpath ~/work/bioformats/artifacts
runxunit components/formats-gpl/test/matlab
```
works (should it be obvious to a BF dev that you need to add those paths?)

Specifying a particular class fails:
```
>> runxunit components/formats-gpl/test/matlab TestBfsave
Error using TestBfsave
Method 'TestBfsave' is not defined for class 'TestBfsave' or is removed
from MATLAB's search path.

Error in TestSuite.fromTestCaseClassName (line 180)
                    suite.add(feval(class_name, method_name));


Error in TestSuite.fromName (line 211)
                suite = TestSuite.fromTestCaseClassName(name);


Error in runxunit (line 100)
            suite.add(TestSuite.fromName(name_list{k}));
 ```

File: docs/sphinx/developers/commit-testing.txt (hunk 3 above)
Comment (manics) [inline docs/sphinx/developers/commit-testing.txt]: If `psexton/matlab-xunit` works for all Matlab versions you could make that the only supported one.

Comment (sbesson) [inline docs/sphinx/developers/commit-testing.txt]: That's definitely what I was thinking of (and migrate to MATLAB built-in `unittest` framework in the mid-term - created a card for that).

File: docs/sphinx/developers/commit-testing.txt [hunk 4]
```diff
@@ -98,3 +101,52 @@ If Ant reports that the build was successful, then there is nothing that
...
+To run these tests, you will need to download `matlab-xunit <https://github.com/psexton/matlab-xunit>`_ which is based on the xUnit
+Framework and adds JUnit-compatible XML output. First add this package together
+with the Bio-Formats MATLAB to your MATLAB path:
+
+.. code-block:: matlab
+
+  % Add the xUnit toolbox to the MATLAB path
+  addpath('/path/to/matlab-xunit');
+  % Add the Bio-Formats dependencies from the source code
+  addpath('components/formats-gpl/matlab');
```
Comment (manics) [inline docs/sphinx/developers/commit-testing.txt]: This needs to be an absolute path to work with the later `cd` commands.

File: docs/sphinx/developers/commit-testing.txt [hunk 5]
```diff
@@ -98,3 +101,52 @@ If Ant reports that the build was successful, then there is nothing that
...
+  addpath('components/artifacts');
```
Comment (manics) [inline docs/sphinx/developers/commit-testing.txt]: `artifacts` is top-level.

**Comment by manics** [comment]
All looks good now

File: docs/sphinx/developers/commit-testing.txt [hunk 6]
```diff
@@ -98,3 +101,54 @@ If Ant reports that the build was successful, then there is nothing that
 you need to do. Otherwise, it is helpful if you can provide the command,
 branch name, number of failures at the bottom of the Ant output, and the
 :file:`bio-formats-software-test-\*.log` file.
+
+MATLAB tests
+------------
+.. _MATLAB xUnit Test Framework: http://uk.mathworks.com/matlabcentral/fileexchange/22846-matlab-xunit-test-framework
```
Comment (manics) [inline docs/sphinx/developers/commit-testing.txt]: Just realised this link needs to be updated to https://github.com/psexton/matlab-xunit

**Comment by sbesson** [comment]
Pushed.

**Comment by manics** [comment]
Looks good

**Pull request merged**